### PR TITLE
Move handle resolution to .well-known

### PR DESCRIPTION
**Pull request by dholms**
This moves handle resolution to `.well-known`

Specifically, a handle is resolved by calling `GET http(s)://${handle}/.well-known/atproto-did/${handle}`

The expected response is `text/plain` and simply contains the associated DID

Closes https://github.com/bluesky-social/atproto/issues/969

### CODE REVIEW

**Comment by bnewbold** [comment]
It feels to me like this is generic to DIDs in general, not just atproto, and we could represent that in the path name. Eg, `/.well-known/did/${handle}` or `/.well-known/handle-did/${handle}`.

Is it a requirement that the handle in the hostname match the handle in the path?

**Comment by bnewbold** [comment]
I actually really like the sub-directory thing, makes it quite easy to do virtual hosting with a bunch of domains and DIDs using something like nginx

**Comment by dholms** [comment]
While it is generic to dids, I think it's useful for different ecosystems to be able to differentiate the dids/handles associated with each.

For instance, with the DNS records we do
`TXT.  _atproto.${handle}   "did=did:...."`

> Is it a requirement that the handle in the hostname match the handle in the path?

I don't have a super strong opinion on this. But I do think it's reasonable to be able to ask an authority what a subdomain resolves to. ie: asking `bsky.social` what `dholms.bsky.social` resolves to. 

Though canonically, you're encouraged to ask the hostname that matches the handle

**Comment by dholms** [comment]
I was trying to match our DNS semantics with the pathing here. maybe `.well-known/atproto-did/${handle}`?

File: packages/identifier/tests/resolve.test.ts [hunk 1]
```diff
@@ -48,12 +48,12 @@ describe('handle resolution', () => {
   })
 
   it('handles a bad DNS resolution', async () => {
-    await expect(resolveDns('bad.test')).rejects.toThrow(NoHandleRecordError)
+    const did = resolveDns('bad.test')
```
Comment (Evalprime) [inline packages/identifier/tests/resolve.test.ts:51]: should this be an await statement? it's failing tests

**Review by Evalprime — COMMENTED** [review]
line 51 in the resolve test needs an await statement

File: packages/identifier/tests/resolve.test.ts (hunk 1 above)
Comment (dholms) [reply packages/identifier/tests/resolve.test.ts]: yup 👍

**Comment by dholms** [comment]
Yeah I think we'll need to wait until we have an actual spec before we can register there

**Comment by Evalprime** [comment]
is the spec documented and will the lexicon work with the site?

File: lexicons/com/atproto/identity/resolveHandle.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
       "description": "Provides the DID of a repo.",
       "parameters": {
         "type": "params",
+        "required": ["handle"],
         "properties": {
           "handle": {
```
Comment (pfrazee) [inline lexicons/com/atproto/identity/resolveHandle.json:12]: should probs drop the "if not specified" bit of the description (at least it was still in the comment)

**Review by pfrazee — APPROVED** [review]
Yep I'm good with this

**Review by pfrazee — APPROVED** [review]
Yep I'm good with this

File: packages/identity/src/handle/index.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      return undefined
+    }
+    return found[0].slice(PREFIX.length)
+  }
+
+  async resolveHttp(
+    handle: string,
+    signal?: AbortSignal,
+  ): Promise<string | undefined> {
+    const url = `https://${handle}/.well-known/atproto-did`
```
Comment (devinivy) [inline packages/identity/src/handle/index.ts:51]: Not a must-have, but we might be able to squeeze a little more safety out of this by ensuring any funny business in `handle` gets ignored (i.e. ensure it is treated as a hostname, no additional path parts in there).  One approach to that:
```suggestion
    const url = new URL('/.well-known/atproto-did', `https://${handle}`).href
```

File: packages/pds/Dockerfile [hunk 4]
```diff
@@ -10,8 +10,8 @@ COPY ./packages/aws ./packages/aws
 COPY ./packages/common ./packages/common
 COPY ./packages/common-web ./packages/common-web
 COPY ./packages/crypto ./packages/crypto
-COPY ./packages/did-resolver ./packages/did-resolver
 COPY ./packages/identifier ./packages/identifier
+COPY ./packages/identity ./packages/identity
```
Comment (devinivy) [inline packages/pds/Dockerfile:14]: Thanks for remembering this 👍

**Review by devinivy — APPROVED** [review]
Good stuff!

File: packages/identity/src/handle/index.ts (hunk 3 above)
Comment (dholms) [reply packages/identity/src/handle/index.ts:51]: nice i like 👍

**Comment by AaronGoldman** [comment]
For some reason I was expecting
```
https://${handle}/.well-known/_atproto -> did=${did}
```
To match the DNS form as closely as possible 🤷‍♂️

**Review by devinivy — APPROVED** [review]

**Comment by zfaizal2** [comment]
Curious, how are you configuring your domains to support wildcard subdomains (ex. `foobar.bsky.social/.well-known/atproto-did`)

**Pull request merged**